File: CodeWars/Fundamentals/k-Primes/Java/src/KPrimes.java
import java.util.stream.LongStream;

public class KPrimes {

    public static long nPrime(long n) {
        int result = 0;

        for (long i = 2 ; i < Math.sqrt(n)+1 ; ++i) {
            while (n%i == 0) {
                result++;
                n /= i;
            }
        }

        if (n > 1)
            result++;

        return result;
    }

    public static long[] countKprimes(int k, long start, long end) {
        return LongStream.range(start, end+1).filter(x -> nPrime(x) == k).toArray();
    }

    public static int puzzle(int s) {
        int result = 0;

        for (long i : countKprimes(1, 1, s)) {
            for (long j : countKprimes(3, 1, s)) {
                for (long k : countKprimes(7, 1, s)) {
                    if (i + j + k == s)
                        result++;
                }
            }
        }

        return result;
    }
}

File: CodeWars/Fundamentals/k-Primes/k-Primes.py
import unittest
from math import sqrt


def n_prime(n):
    result = 0

    for i in range(2, int(sqrt(n))+1):
        while n % i == 0:
            result += 1
            n /= i

    return (result + 1) if n > 1 else result


def count_kprimes(k, start, end):
    return [x for x in range(start, end+1) if n_prime(x) == k]


def puzzle(s):
    pass


class TestExercise(unittest.TestCase):
    def test_count_kprimes(self):
        a = [1020, 1026, 1032, 1044, 1050, 1053, 1064, 1072, 1092, 1100]
        self.assertEqual(count_kprimes(5, 1000, 1100), a)
        a = []
        self.assertEqual(count_kprimes(12, 100000, 100100), a)

    def test_puzzle(self):
        self.assertEqual(puzzle(100), 0)
        self.assertEqual(puzzle(144), 0)
        self.assertEqual(puzzle(143), 2)


if __name__ == '__main__':
    unittest.main()
